compiler/grainlsp/lenses.re

```diff
@@ -0,0 +1,87 @@
+open Grain_typed;
+open Grain_diagnostics;
+
+let get_signature_from_statement =
+    (~log, stmt: Grain_typed__Typedtree.toplevel_stmt) =>
+  switch (stmt.ttop_desc) {
+  | TTopImport(import_declaration) => None
+  | TTopForeign(export_flag, value_description) => None
+  | TTopData(data_declarations) =>
+    switch (data_declarations) {
+    | [] => None
+    | _ =>
+      let decls =
+        List.fold_left(
+          (acc, decl: Typedtree.data_declaration) =>
+            (acc == "" ? "" : acc ++ ", ")
+            ++ (
+              switch (decl.data_kind) {
+              | TDataVariant(_) => "enum " ++ decl.data_name.txt
+              | TDataRecord(_) => "record " ++ decl.data_name.txt
+              | TDataAbstract => "abstract" // not sure about this one
```

Comment:
This is either a type alias or abstract type, depending on whether or not it has a type manifest